examples/player/simid_player.js

```diff
@@ -197,33 +220,100 @@ class SimidPlayer {
   }
 
   /**
-   * Returns the dimensions of an element within the player div.
+   * Returns the full dimensions of an element within the player div.
+   * These dimensions will equal the full size of the player.
    * @return {!Object}
    */
-  getDimensions(elem) {
-    // The player div wraps all elements and is used as the offset.
-    const playerDiv = document.getElementById('player_div');
-    const playerRect = playerDiv.getBoundingClientRect();
+  getFullVideoDimensions(elem) {
     const videoRect = elem.getBoundingClientRect();
+
     return {
-      'x' : videoRect.x - playerRect.x,
-      'y' : videoRect.y - playerRect.y,
+      'x' : 0,
+      'y' : 0,
       'width' : videoRect.width,
       'height' : videoRect.height,
       // TODO: This example does not currently support transition duration.
       'transitionDuration': 0
     };
   }
 
+  /**
+   * Returns the specified dimensions of the non-linear creative.
+   * @return {!Object}
+   */
+  getNonLinearDimensions() {
+    const x_val = document.getElementById('x_val').value;
+    const y_val = document.getElementById('y_val').value;
+    const width = document.getElementById('width').value;
+    const height = document.getElementById('height').value;
+
+    return {
+      'x' : x_val,
+      'y' : y_val,
+      'width' : width,
+      'height' : height,
+      // TODO: This example does not currently support transition duration.
+      'transitionDuration': 0
+    };
+  }
+
+  /**
+   * Displays the non-linear creative with the specified size
+   * on top of the video content.
+   */
+  displayNonLinearCreative_() {
+    const dimensions = this.getNonLinearDimensions();
+    
+    this.simidIframe_.style.height = dimensions.height;
+    this.simidIframe_.style.width = dimensions.width;
+    this.simidIframe_.style.left = `${dimensions.x}px`;
+    this.simidIframe_.style.top = `${dimensions.y}px`;
+
+    this.simidIframe_.style.position = "absolute";
+
+    this.contentVideoElement_.play();
+  }
+
+  /**
+   * Allows users to request resizing of the creative
+   * @param {!Object} incomingMessage Message sent from the 
+   *   creative to the player
+   */
+  onRequestResize(incomingMessage) {
+    const x_val  = incomingMessage.args['x_val'];
+    const y_val  = incomingMessage.args['y_val'];
+    const width  = incomingMessage.args['width'];
+    const height  = incomingMessage.args['height'];
+
+    const resizeDimensions = {
+      'x' : x_val,
+      'y' : y_val,
+      'width' : width,
+      'height' : height,
+    }
+
+    if (this.isLinearAd_) {
+      console.log("Cannot resize linear ad");
+      return;
+    }
+
+    this.simidIframe_.style.height = resizeDimensions.height;
```

Comment:
Youre probably going to want to validate these values somewhere.  Make sure that they are actually numbers, and that they are valid numbers (not infinity, not negative, not bigger than the player etc.)

Reply:
we do- Kaleab put that in a different PR